src/image_intel.cpp

```diff
@@ -40,6 +44,64 @@ static void setBootAddress(size_t address)
     std::ignore = exec("fw_setenv bootcmd bootm %08x", address);
 }
 
+using FileSystem = std::string;
+using MountPoint = std::string;
+using MountPoints = std::vector<std::pair<FileSystem, MountPoint>>;
+
+static MountPoints getMountPoints()
+{
+    static const std::regex MountPointRecord("^(mtd:\\w+)\\s+([^\\s]+)\\s+.*$");
+    MountPoints mtdPartitions;
+    std::ifstream mounts("/proc/mounts");
+
+    if (mounts.is_open())
+    {
+        std::string line;
+        while (std::getline(mounts, line))
+        {
+            std::smatch match;
+            if (std::regex_match(line, match, MountPointRecord))
+            {
+                mtdPartitions.emplace_back(
+                    std::make_pair(match[1].str(), match[2].str()));
+            }
+        }
+
+        mounts.close();
+    }
+
+    return mtdPartitions;
+}
+
+static void unmountFilesystem(const std::string& filesystem)
+{
+    // Some systemd services may to occupate the RW partition
+    // and make delay up to 20 seconds. We should to wait them
+    // befor throw an error.
+    constexpr auto maxTryings = 40; // Max number of tryings to umount
```

Comment:
`maxTries`, `// Max number of tries to unmount`

Reply:
fixed